sound/soc/bcm/rpi-simple-soundcard.c

```diff
@@ -354,16 +354,46 @@ static struct snd_rpi_simple_drvdata drvdata_hifiberry_dac = {
 	.dai       = snd_hifiberry_dac_dai,
 };
 
+SND_SOC_DAILINK_DEFS(hifiberry_dac8x,
+	DAILINK_COMP_ARRAY(COMP_EMPTY()),
+	DAILINK_COMP_ARRAY(COMP_CODEC("snd-soc-dummy", "snd-soc-dummy-dai")),
+	DAILINK_COMP_ARRAY(COMP_EMPTY()));
+
 static int hifiberry_dac8x_init(struct snd_soc_pcm_runtime *rtd)
 {
 	struct snd_soc_dai *codec_dai = asoc_rtd_to_codec(rtd, 0);
+	struct snd_soc_card *card = rtd->card;
+	struct gpio_desc *gpio_desc;
+	bool has_adc;
 
-	/* override the defaults to reflect 4 x PCM5102A on the card
-	 * and limit the sample rate to 192ksps
-	 */
+	/* Configure the codec for 8 channel playback */
 	codec_dai->driver->playback.channels_max = 8;
 	codec_dai->driver->playback.rates = SNDRV_PCM_RATE_8000_192000;
 
+	/* Activate capture based on ADC8x detection */
+	gpio_desc = devm_gpiod_get(card->dev, "hasadc", GPIOD_IN);
+	if (IS_ERR(gpio_desc)) {
+		dev_err(card->dev, "Failed to get GPIO: %ld\n", PTR_ERR(gpio_desc));
+		return PTR_ERR(gpio_desc);
+	}
+
+	has_adc = !gpiod_get_value(gpio_desc);
```

Comment:
Oh - then you invert it. Wouldn't it be more accurate, more obvious and generally cleaner to declare the signal as active low in the overlay, removing the need for this negation?

Reply:
Yes, you are absolutely right. It's just that I don't like 'negative' logic ... I'll wait for the checks to complete and change it.

Reply:
The checks are going to pass - I've run a cut-down version on a faster machine already.